### PR TITLE
React Group Project

### PR DESCRIPTION
**Pull request by anyars-encarta**
In this project:
- [ ] We created a Space Travelers' Hub app suing React and Redux
- [ ]  Created a Rockets page that fetches rockets data from an API and display them the UI
- [ ]  Created Missions page that fetches missions data from an API and renders them on the UI
- [ ]  Created My Profile page that display reserved rockets and joined missions
- [ ]  Created a button on the Rockets page to reserve or cancel a Rocket reservation
- [ ]  Created a button to Join or leave a mission
- [ ]  Reserved Rockets are displayed on the My profile page
- [ ]  We wrote unit tests for all components using React Testing Library and Jest

### CODE REVIEW

**Comment by Reem-lab** [comment]
Hello @anyars-encarta could you check your slack?

File: src/components/My-profile.jsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+const selectReservedMissions = createSelector(
+  [selectMissions],
+  (missions) => missions.filter((mission) => mission.reserved),
+);
+
+const selectReservedRockets = createSelector(
+  [selectRockets],
+  (rockets) => rockets.filter((rocket) => rocket.reserved),
+);
+
```
Comment (Reem-lab) [inline src/components/My-profile.jsx:20]: - [ ] Great job so far working on the project I think you made a huge effort on it and it is really appreciated but I think you have a small bug here so when displaying the missions and rockets on my profile page when I try to reserve a rocket and go to the profile page I can see it but when then come back to the rockets page I could see the reserve is gone but it should still there till the user cancel it by himself the same for missions so kindly fix it to still show that it is reserved 🙏🏻✨😄✅🔥


- Here's a video about what I am asking for ⬇


https://github.com/anyars-encarta/react-group-project/assets/58553711/514d96b8-7d5e-45c4-9419-f668b6d062c9

File: src/__tests__/missions.test.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    description: 'Mission Mission, Mission Mission Space',
+  },
+  {
+    mission_id: 'mission2',
+    mission_name: 'Mission Apolo 2',
+    description: 'Mission Mission, Mission Mission Space 2',
+  },
+  {
+    mission_id: 'mission3',
+    mission_name: 'Mission Apolo 3',
```
Comment (Reem-lab) [inline src/__tests__/missions.test.js:27]: - [ ] Great job working on the tests but I think you have failed tests I could show you the error message and please try to check on the issue and solve it ⬇🔥⭐It's good practice to have passed tests for all the methods to ensure that they work as expected.. Kindly make sure to test it 💪

1-
image

2-
image

3- 
image

4-

image

**Review by Reem-lab — CHANGES_REQUESTED** [review]
# Hi @anyars-encarta and @BiancaMe, 👋🏻👋🏻

Great job so far! 💯🤩🚀
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- No Linters error ✨
- Followed Gitflow 🔥
- Great working on the functionality 💯
- Great set up the redux slices and store🤩🤩
- Professional README 💯
- Kudos for adding the required tests✨
- Great job working on the design 🔥
- Professional README ⭐🔥
- Amazing for giving the reviews to each other 💯💯

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/My-profile.jsx (hunk 1 above)
Comment (anyars-encarta) [reply src/components/My-profile.jsx:20]: Thank you for that observation. We will look into it

File: src/__tests__/missions.test.js (hunk 2 above)
Comment (anyars-encarta) [reply src/__tests__/missions.test.js:27]: This was very unexpected. We shall look into it asap

**Pull request merged**